### DISCUSSION TITLE
SQL问题

### DISCUSSION THREAD

**Discussion by PureDimension** [liusy58/NJU_AdvancedProgramming_SP22#131] [Q&A] [answered]
![Screenshot_20220314_111612](https://user-images.githubusercontent.com/100058437/158105819-2498f082-1b88-4ccd-8f57-2fc4234615b4.jpg)
可不可以直接用WHERE seller=... 区分，感觉还单独传角色作为参数好傻（

**Comment by binbin00** [comment, marked as answer]
不可

**Reply by PureDimension** [reply]
行（

**Reply by PureDimension** [reply]
还有想问一下，比如我在执行SQL中update之前确认信息的环节，我确认信息的方法是使用额外的SELECT指令来寻找信息，那么我在commands.txt中允许存在额外的SQL执行记录（比如update前的select）吗？还是必须要另外用别的方法实现

**Reply by binbin00** [reply]
请不要把问题复杂化，确认信息直接去遍历相应的表即可。

**Reply by cqjjjzr** [reply]
这是还要搓一套apache shiro一样的权限机制出来吗（

参照 https://github.com/liusy58/NJU_AdvancedProgramming_SP22/discussions/27

**Reply by PureDimension** [reply]
> 这是还要搓一套apache shiro一样的权限机制出来吗（
> 
> 参照 #27

其实我最后的解决办法是把完整表格拿回来以后自己写个工具函数裁剪，相当敷衍（
